model/converter/thrift/zipkin/to_domain.go

```diff
@@ -76,39 +76,62 @@ func ToDomain(zSpans []*zipkincore.Span) (*model.Trace, error) {
 // A valid model.Span is always returned, even when there are errors.
 // The errors are more of an "fyi", describing issues in the data.
 // TODO consider using different return type instead of `error`.
-func ToDomainSpan(zSpan *zipkincore.Span) (*model.Span, error) {
+func ToDomainSpan(zSpan *zipkincore.Span) ([]*model.Span, error) {
 	return toDomain{}.ToDomainSpan(zSpan)
 }
 
 type toDomain struct{}
 
 func (td toDomain) ToDomain(zSpans []*zipkincore.Span) (*model.Trace, error) {
 	var errors []error
-	trace := &model.Trace{}
 	processes := newProcessHashtable()
+	trace := &model.Trace{}
 	for _, zSpan := range zSpans {
-		jSpan, err := td.ToDomainSpan(zSpan)
+		jSpans, err := td.ToDomainSpan(zSpan)
 		if err != nil {
 			errors = append(errors, err)
 		}
-		// remove duplicate Process instances
-		jSpan.Process = processes.add(jSpan.Process)
-		trace.Spans = append(trace.Spans, jSpan)
+		for _, jSpan := range jSpans {
+			// remove duplicate Process instances
+			jSpan.Process = processes.add(jSpan.Process)
+			trace.Spans = append(trace.Spans, jSpan)
+		}
 	}
 	return trace, multierror.Wrap(errors)
 }
 
-func (td toDomain) ToDomainSpan(zSpan *zipkincore.Span) (*model.Span, error) {
-	jSpan := td.transformSpan(zSpan)
+func (td toDomain) ToDomainSpan(zSpan *zipkincore.Span) ([]*model.Span, error) {
+	jSpans := td.transformSpan(zSpan)
 	jProcess, err := td.generateProcess(zSpan)
-	jSpan.Process = jProcess
-	return jSpan, err
+	for _, jSpan := range jSpans {
+		jSpan.Process = jProcess
+	}
+	return jSpans, err
+}
+
+// isClientAndServerSpan determines whether span represents both client and server processing
+func (td toDomain) isClientAndServerSpan(zSpan *zipkincore.Span) bool {
+	var csOk bool
+	var srOk bool
+	for _, ann := range zSpan.Annotations {
+		if ann.Value == "cs" {
+			csOk = true
+		} else if ann.Value == "sr" {
+			srOk = true
+		}
+		if csOk && srOk {
+			return true
+		}
+	}
+	return false
 }
 
 // transformSpan transforms a zipkin span into a Jaeger span
-func (td toDomain) transformSpan(zSpan *zipkincore.Span) *model.Span {
+func (td toDomain) transformSpan(zSpan *zipkincore.Span) []*model.Span {
 	tags := td.getTags(zSpan.BinaryAnnotations, td.isSpanTag)
-	if spanKindTag, ok := td.getSpanKindTag(zSpan.Annotations); ok {
+	var spanKindTag model.KeyValue
```

Comment:
I don't think you need to capture this tag, the model span has helper methods like `span.IsRPCClient()`